pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -581,11 +582,17 @@ impl InMemoryLayer {
                     for (lsn, pos) in vec_map.as_slice() {
                         cursor.read_blob_into_buf(*pos, &mut buf, &ctx).await?;
                         let will_init = Value::des(&buf)?.will_init();
-                        let res;
-                        (buf, res) = delta_layer_writer
-                            .put_value_bytes(Key::from_compact(*key), *lsn, buf, will_init, &ctx)
+                        let (tmp, res) = delta_layer_writer
+                            .put_value_bytes(
+                                Key::from_compact(*key),
+                                *lsn,
+                                std::mem::take(&mut buf).slice_len(),
```

Comment:
nit: this is like the optional trick we've used in other places, but without the optional, right? Any reason for doing it differently here?

Reply:
No reason. Personally I like the Optional variant better because it gives a hard panic if we're not reusing the buffer

Reply:
But, turns out the compiler is smart enough b55a70ab1b0538d9f05654a898afe155235eae45